crates/guest-rust/rt/src/async_support/future_support.rs

```diff
@@ -1,50 +1,53 @@
-extern crate std;
+//! Runtime support for `future<T>` in the component model.
+//!
+//! TODO:
+//!
+//! * leaking requires owned values
+//! * owned values means we can't return back sent items
+//! * intimately used in implementation details.
```

Comment:
I do want to expand long-form documentation here about all the concerns around leaks/ownership/io_uring-lookalike/etc.